src/components/PopOver/compositions/usePopOverGroup.ts

```diff
@@ -0,0 +1,26 @@
+export type PopOverOpenMethod = () => void
+export type PopOverCloseMethod = () => void
+
+export type UsePopOverGroup = () => {
+  setGroupCloseMethod: (groupId: string, close: PopOverCloseMethod) => void,
+}
+
+function factory(): UsePopOverGroup {
```

Comment:
I'm curious why you chose to name this function `factory` rather than exporting `usePopOverGroup` directly?

Reply:
This is a pattern I've used in quite a few compositions to accomplish encapsulation of a shared state. Basically write a factory method that returns the actual exported composition (bottom of the file). Which allows the composition to share some values across all composition calls (in this case the `groups` const and the `setGroupCloseMethod` function).

Reply:
Ahh, gotchya. Rather than creating multiple instances of the function. Makes sense, thank you!